pkg/util/k8sutil/k8sutil.go

```diff
@@ -365,11 +365,19 @@ func newEtcdPod(m *etcdutil.Member, initialCluster []string, clusterName, state,
 				Name:  "check-dns",
 				// In etcd 3.2, TLS listener will do a reverse-DNS lookup for pod IP -> hostname.
 				// If DNS entry is not warmed up, it will return empty result and peer connection will be rejected.
+				// In some cases the DNS is not created correctly so we time out after 10 minutes in this case.
 				Command: []string{"/bin/sh", "-c", fmt.Sprintf(`
+					TIMEOUT_READY=%d
 					while ( ! nslookup %s )
 					do
-						sleep 2
-					done`, m.Addr())},
+						# If TIMEOUT_READY is 0 we should never time out and exit 
+						if [ $(( TIMEOUT_READY-- )) -eq 0 ];
+				            then
+				                echo "Timed out waiting for DNS entry"
+				                exit 1
+				            fi
+						sleep 1
+					done`, m.Addr(), cs.Pod.DNSTimeoutInSecond)},
```

Comment:
the ordering of the args`fmt.Sprintf("%d %s", m.Addr(),cs.Pod.DNSTimeoutInSecond)` doesn't seem right. Should it be `fmt.Sprintf("%d %s", cs.Pod.DNSTimeoutInSecond, m.Addr())`?

Reply:
You're right! Let me fix this